dashbuilder/dashbuilder-client/dashbuilder-renderers/dashbuilder-renderer-c3/src/main/java/org/dashbuilder/renderer/c3/client/C3DisplayerView.java

```diff
@@ -95,4 +102,29 @@ public void setSize(int width, int height) {
         this.height = height;
     }
     
-}
+    public void setResizable(boolean resizable, int maxWidth, int maxHeight) {
+        if (resizable) {
```

Comment:
I think we should throw a runtime exception if we try to setReziable in a not resizable view.

Reply:
Hello,

Thanks for your comments.

> I think we should throw a runtime exception if we try to setReziable in a not resizable view.

This should never happen because each displayer must provide if it supports a certain property or not (e.g. resizable). If we don't say that certain view does not support the property then it will always be false, meaning that the setResizable method will never be called;

> The current setResize api feels strange.

I do agree with you, I was making some changes when it is not resizable (clean the width when it is not resizable), but it does not seem to be necessary. Will change the API.


> when the resize is disabled I would disable the input fields for max width / max height in the UI to avoid confusion about why they don't have any effect. Even better would be to hide the width/height when the resize checkbox is unchecked (but you can ignore this point if this would be hard to implement in the current displayer settings).
> I would rename the labels from width / height to max width / max height to avoid confusion like "Why do I have to set width if it's resizable?"

Although I agree with both improvements, currently it is indeed complicate to implement this on specific ticket IMO. It would run out of the main scope from the original JIRA. But I can create new JIRAs for future investigation.